### PR TITLE
Support DynamicConfiguration in feature gate

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve? 
Close https://github.com/pingcap/tidb-operator/issues/1662
Close https://github.com/pingcap/tidb-operator/issues/2511

Add a featureGate to control the DynamicConfiguration. For an existed TidbCluster, it should both enable DynamicConfiguration feature and ConfigUpdateStrategyRollingUpdate to enable DynamicConfiguration as it need update the tikv cluster.

Manually test:

```sh
mysql> select * from CLUSTER_INFO;
+------+---------------------------------------------------+---------------------------------------------------+--------------------+------------------------------------------+----------------------+---------------+
| TYPE | INSTANCE                                          | STATUS_ADDRESS                                    | VERSION            | GIT_HASH                                 | START_TIME           | UPTIME        |
+------+---------------------------------------------------+---------------------------------------------------+--------------------+------------------------------------------+----------------------+---------------+
| tidb | basic-tidb-0.basic-tidb-peer.tidb-admin.svc:4000  | basic-tidb-0.basic-tidb-peer.tidb-admin.svc:10080 | 5.7.25-TiDB-v4.0.0 | 689a6b6439ae7835947fcaccf329a3fc303986cb | 2020-06-08T08:59:05Z | 1m13.45812s   |
| pd   | basic-pd:2379                                     | basic-pd:2379                                     | 4.1.0-alpha        | 56d4c3d2237f5bf6fb11a794731ed1d95c8020c2 | 2020-06-08T08:57:56Z | 2m22.4581508s |
| tikv | basic-tikv-0.basic-tikv-peer.tidb-admin.svc:20160 | basic-tikv-0.basic-tikv-peer.tidb-admin.svc:20180 | 4.1.0-alpha        | e3058403a0fc9a96870882bf184ac075223b4642 | 2020-06-08T08:58:30Z | 1m48.4581635s |
+------+---------------------------------------------------+---------------------------------------------------+--------------------+------------------------------------------+----------------------+---------------+


```

To enable DynamicConfiguration, we need tikv add advertise-status-addr and let tidb use it instead of status-addr.

ref https://github.com/tikv/tikv/pull/7944

### Does this PR introduce a user-facing change?:

```release-note
Support `DynamicConfiguration` in `TiDB Operator` feature gates.
```

### CODE REVIEW

File: charts/tidb-operator/values.yaml [hunk 1]
```diff
@@ -35,6 +35,7 @@ features: []
 # - AdvancedStatefulSet=false
 # - StableScheduling=true
 # - AutoScaling=false
+# - DynamicConfiguration=true
```
Comment (DanielZhangQD) [inline charts/tidb-operator/values.yaml:38]: We should control this behavior per TidbCluster instead of a global switch.

Comment (Yisaer) [reply charts/tidb-operator/values.yaml:38]: make sense

**Pull request merged**